src/ClimaLSM.jl

```diff
@@ -103,22 +104,89 @@ Returns the component names of the `land` model, by calling
+end
+
+function prognostic_types(land::AbstractLandModel)
+    components = land_components(land)
+    prognostic_list = map(components) do model
+        prognostic_types(getproperty(land, model))
+    end
+    return NamedTuple{components}(prognostic_list)
+end
+
+function auxiliary_vars(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_vars(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_vars(land),
+    ))
+end
+
+function auxiliary_types(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_types(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_types(land),
+    ))
+end
+
 """
-    initialize_interactions(land::AbstractLandModel) end
+   interaction_vars(m::AbstractModel)
 
-Initializes interaction variables, which are a type of auxiliary
-variable, to empty objects of the correct type
-for the model. 
+Returns the interaction variable symbols for the model in the form of a tuple.
+"""
+interactions_vars(m::AbstractLandModel) = ()
+
+"""
+   interaction_types(m::AbstractModel)
+
+Returns the shared interaction variable types for the model in the form of a tuple.
+"""
+interactions_types(m::AbstractLandModel) = ()
+
+"""
+   interaction_spaces(m::AbstractModel)
 
-This function should be called during `initialize_auxiliary`,
-and the method will change depending on the type of land model,
-and potentially due to the type of the component models. 
+Returns the symbol of a model on the interaction space in the form of a tuple.
 
-This is a stub which is extended to concrete LSM models.
+For example, for a boundary interaction term, one would specify a model with a
+domain consisting of the boundary space. Most commonly, these are surface models.
+"""
+interactions_spaces(m::AbstractLandModel) = ()
+
+"""
+    initialize_interactions(land::AbstractLandModel) end
 
+Initializes interaction variables, which are a type of auxiliary
+variable, to empty objects of the correct type for the model. 
+
+Interaction variables are specified by `interaction_vars`, their types
+by `interaction_types`, and their spaces by `interaction_spaces`. 
+This function should be called during `initialize_auxiliary` step.
 """
-function initialize_interactions(land::AbstractLandModel) end
+function initialize_interactions(land::AbstractLandModel, land_coords)
+    vars = interactions_vars(land)
+    types = interactions_types(land)
+    domains = interactions_spaces(land)
+    interactions = map(zip(types, domains)) do (T, domain)
+        zero_instance = zero(T)
+        map(_ -> zero_instance, getproperty(land_coords, domain))
+    end
 
+    return NamedTuple{vars}(interactions)
+end
```

Comment:
Generic initialize - we may be able to unify this with `initialize_vars` but that would require changing the output type of that (currently a `FieldVector`). We would probably return a pair of `keys`, `zero_instances` where `keys` are the variable names and the `zero_instances` are the initialized fields. Then we can package that into a `NamedTuple` or `FieldVector`.

Reply:
Right - this doesnt need to return something can be run in standalone mode. I think I'm fine with it being its own thing & not part of a single `init_vars`  - this is all part of the LSM `initialize` method, which we are specializing for (it's different from the `initialize` of `AbstractModel`, though it make use of some of the generic methods that `init` of abstractmodels employs). We can also change the names to be more clear? `initialize_vars` vs `initialize_vectors`? Not sure if that is necessary but it is an option.

Reply:
I think it makes sense to unify things, but maybe in a following PR. A benefit to returning the `keys` and `zero_instances` is that we could then have flexibility in how we package these states (FieldVectors, NamedTuples).